File: Metode-Avansate-de-Programare/practic/PracticAjutor/src/main/java/ubb/map/practicajutor/repository/IRepository.java
package ubb.map.practicajutor.repository;


import ubb.map.practicajutor.domain.Entity;

import java.util.Optional;

public interface IRepository<ID, E extends Entity<ID>> {
    Optional<E> findOne(ID id);

    Iterable<E> findAll();

    Optional<E> save(E entity);

    Optional<E> delete(ID id);

    Optional<E> update(E entity);
}

File: Medii-de-Proiectare-si-Programare/App/App/Repositories/IRepository.cs
namespace App.Repositories;

public interface IRepository<TId, TEntity>
{
    TEntity? FindOne(TId id);
    IEnumerable<TEntity> FindAll();
    TEntity? Save(TEntity entity);
    TEntity? Delete(TId id);
    TEntity? Update(TEntity entity);
}